### DISCUSSION TITLE
Revalidate create too many MySQL connections ?

### DISCUSSION THREAD

**Discussion by mdcarter** [vercel/next.js#15937] [Help] [answered]
I have a webapp hosted on Vercel that is using Next.js `getStaticPaths` and `getStaticProps` methods.
On build (and on runtime via the `revalidate: 10` option) I make a call to my API (also hosted on Vercel) that connects to a MySQL server.

The issue here is when I go to a page that is statically generated, Next.js try to rebuild it in the background (to keep it fresh) and I end up instantly with more than 50 connections to my database. (I have around 80 statically generated pages on the site, around 50 for a certain url and 30 for another)

<img width="480" alt="Screenshot 2020-08-05 at 16 41 06" src="https://user-images.githubusercontent.com/67362/89426448-76d58080-d73a-11ea-8e4c-16ef63ff11d3.png">

If I continue to browse, the connections keep piling up until I reach the MySQL connections limit (152 here) and all the pages builded are now 404 (the API can't respond data, just errors).

Do you have any idea where the problem could arise from ?
Why so many connections are created ?

Thank you in advance, I would be sad to give up statically generated pages that are a big boost of performance :(

**Comment by paulogdm** [comment, marked as answer]
It would be good to consider https://github.com/jeremydaly/serverless-mysql so you can optimize some aspects of your code. 

Can you post a code snippet here?

**Reply by mdcarter** [reply]
I did not know about this, switched to this and our number of MySQL connexions seems to hover around 120, which is good !
I'll make more tests but this seems to resolve my issue, a BIG THANK YOU 🙏

**Reply by leerob** [reply]
For others who have found this discussion, check out the latest MySQL guide and example: https://github.com/vercel/next.js/discussions/19162#discussioncomment-137742
